app/jobs/create_group_and_add_members_job.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+##
+# This class is responsible for assigning a conceptually "private" group to the
+# members of of the {Cdl} work.
+#
+# The reason for creating this group is when a person gains access to the CDL 
+# (via the lending application) it is more performant to add the person to a 
+# group than it is to add the person directly to each of the underlying files.
+class CreateGroupAndAddMembersJob < ApplicationJob
+```#
+  RETRY_MAX = 5
+
+  queue_as :default
+
+  def perform(cdl_id, retries = 0)
+    work = Cdl.where(id: cdl_id).first
+    return if work.nil?
+
+    page_count = work.file_sets.first.page_count.first.to_i
+    child_model = work.iiif_print_config.pdf_split_child_model
+    child_works_count = work.members.select { |member| member.is_a?(child_model) }.count
+
+    if page_count == child_works_count
+      group = Hyrax::Group.create(name: work.id)
```

Comment:
Given that we have retry attempts favoring a find or create will allow us to:

1. Avoid name collisions
2. Allow for greater fault tolerance.

```suggestion
      group = Hyrax::Group.find_or_create_by!(name: work.id)
```